comp/snmptraps/config/configimpl/service_mock.go

```diff
@@ -14,11 +14,22 @@ import (
 	"go.uber.org/fx"
 )
 
-func newMockConfig(tc *trapsconf.TrapsConfig, hnService hostname.Component) (trapsconf.Component, error) {
-	host, err := hnService.Get(context.Background())
-	if err != nil {
-		return nil, err
+type dependencies struct {
+	fx.In
+	HostnameService hostname.Component `optional:"true"`
```

Comment:
You should avoid `optional` as there is no error when there is a missing dependency in `hostname.Component `.

Reply:
This is deliberate - tests don't need to include the hostname service unless they need to manipulate the hostname in some way. If a test is missing the dependency on hostname.Component, then `newMockConfig` will use a default hostname.